pom.xml

```diff
@@ -579,6 +553,92 @@ limitations under the License.]]></inlineHeader>
     </plugins>
   </build>
   <profiles>
```

Comment:
shouldn't we have a `default` profile as well?

Reply:
there is no need to declare a 'default' profile.
an also I had issues with having a default configuration for the maven-compiler-plugin.

For instance if you choose to support jdk8 as default, you are not able to make it work with jdk11, because the configurations in the compiler plugin use different tags names and Maven merges the configuration in a way that does not work.
The examples I found in ErrorProne website did not work for me.

So this is the best configuration I found, even if it not awesome